modules/installation-adding-registry-pull-secret.adoc

```diff
@@ -0,0 +1,114 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_disconnected/installing-disconnected-preparations.adoc
+
+[id="installation-adding-registry-pull-secret_{context}"]
+= Adding the registry to your pull secret
+
+Modify your the pull secret for your {product-title} cluster to describe
+your local registry before you install an {product-title} cluster in a
+disconnected environment.
+
+.Prerequisites
+
+* You prepared the bastion host to use in a disconnected installation.
+* You configured a mirror registry to use in your disconnected environment.
+
+
+.Procedure
+
+Complete the following steps on the bastion host:
+
+. Download your `registry.redhat.io` pull secret from the
+link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers]
+page.
+
+. Generate the base64-encoded user name and password or token for your mirror
+registry:
++
+----
+$ echo -n '<user_name>:<password>' | base64 -w0 <1>
+
+BGVtbYk3ZHAtqXs=
+----
+<1> For `<user_name>` and `<password>`, specify the user name and password that
+you configured for your registry.
+
+. Make a copy of your pull secret:
++
+----
+$ cat ./pull-secret.text | jq .  > <path>/pull-secret-2.json <1>
+----
+<1> Specify the path to the folder to store the pull secret in and a name for
+the file.
++
+The contents of the file resemble the following example:
++
+----
+{
+  "auths": {
+    "cloud.openshift.com": {
+      "auth": "b3BlbnNo...",
+      "email": "you@example.com"
+    },
+    "quay.io": {
+      "auth": "b3BlbnNo...",
+      "email": "you@example.com"
+    },
+    "registry.connect.redhat.com": {
+      "auth": "NTE3Njg5Nj...",
+      "email": "you@example.com"
+    },
+    "registry.redhat.io": {
+      "auth": "NTE3Njg5Nj...",
+      "email": "you@example.com"
+    }
+  }
+}
+----
+
+. Edit the new file and add a section that describes your registry to it:
++
+----
+  "auths": {
+...
+    "<bastion_host_name>:5000": { <1>
```

Comment:
Doesn't have to be `:5000`, your local registry could serve HTTPS on the default 443.  You could use `<local_registry_authority>` to lean on [the URI nomenclature][1] or `<local_registry_host_port>` because we don't actually want the authority's optional `userinfo "@"` here.

[1]: https://tools.ietf.org/html/rfc3986#section-3.2

Reply:
@wking, that's fair. Do you have any other guidance for a suggested port? I'd be more likely to use `<local_registry_host_port>` so that I can say to open it [here](https://github.com/openshift/openshift-docs/pull/16678/files#diff-7d4591e9ba80df3ebf307fb24525011bR100).

Reply:
5000 seems popular for "I'm running some dev thing".  For production, I'd expect a registry running on 443, but I don't think these docs are where we want to walk folks through production registries.

Reply:
@wking, we need to provide enough information that they could take this example and configure a production registry. I'll make it generic.

Is there any other information that would be valid in a production environment that we don't have yet?

Reply:
> Is there any other information that would be valid in a production environment that we don't have yet?

If you mirror to an existing, production registry, you probably just need the `oc adm release mirror ...` step and `imageContentSources` block.  You won't need anything for `additionalTrustBundle` to support the mirror registry, because it will already have a cert signed by a RHCOS-trusted CA.  If the mirrored images are public on the mirror registry, you won't need a mirror token in your pull secret.  I dunno, there's not all that much else going on...

Reply:
I'm glad that it's that easy.